network/address_test.go

```diff
@@ -27,6 +29,12 @@ func TestConnType(t *testing.T) {
 }
 
 func TestAddress(t *testing.T) {
+	localhostNetworkAddresses, _ := net.LookupHost("localhost")
+	sort.Strings(localhostNetworkAddresses)
+	facebookNetworkAddresses, _ := net.LookupHost("facebook.com")
+	sort.Strings(facebookNetworkAddresses)
+	googleNetworkAddresses, _ := net.LookupHost("google.com")
```

Comment:
I was just thinking of something more hacky-ish, having a non exported global variable `var res Resolver = NewGolangNetResolver()`. When doing the test, you just set that variable to `Resolver = newTestResolver()`. 
Of course, as @ineiti suggested, we can do that with just a simple `type resolv func(host string) (addrs []string,error)`, I just find less beauty in it, but I'm totally fine with it too.